pop-api/src/v0/assets/fungibles.rs

```diff
@@ -0,0 +1,622 @@
+/// in unspecified format.
+///
+/// # Parameters
+/// - `id` - The ID of the asset.
+/// - `from` - The account from which the tokens are transferred.
+/// - `to` - The recipient account.
+/// - `value` - The number of tokens to transfer.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the transfer fails.
+#[inline]
+pub fn transfer_from(id: AssetId, from: AccountId, to: AccountId, value: Balance) -> Result<()> {
+	build_dispatch(TRANSFER_FROM)
+		.input::<(AssetId, AccountId, AccountId, Balance)>()
+		.output::<Result<()>, true>()
+		.handle_error_code::<StatusCode>()
+		.call(&(id, from, to, value))
+}
+
+/// Approves an account to spend a specified number of tokens on behalf of the caller.
+///
+/// # Parameters
+/// - `id` - The ID of the asset.
+/// - `spender` - The account that is allowed to spend the tokens.
+/// - `value` - The number of tokens to approve.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the approval fails.
+#[inline]
+pub fn approve(id: AssetId, spender: AccountId, value: Balance) -> Result<()> {
+	build_dispatch(APPROVE)
+		.input::<(AssetId, AccountId, Balance)>()
+		.output::<Result<()>, true>()
+		.handle_error_code::<StatusCode>()
+		.call(&(id, spender, value))
+}
+
+/// Increases the allowance of a spender.
+///
+/// # Parameters
+/// - `id` - The ID of the asset.
+/// - `spender` - The account that is allowed to spend the tokens.
+/// - `value` - The number of tokens to increase the allowance by.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the operation fails.
+#[inline]
+pub fn increase_allowance(id: AssetId, spender: AccountId, value: Balance) -> Result<()> {
+	build_dispatch(INCREASE_ALLOWANCE)
+		.input::<(AssetId, AccountId, Balance)>()
+		.output::<Result<()>, true>()
+		.handle_error_code::<StatusCode>()
+		.call(&(id, spender, value))
+}
+
+/// Decreases the allowance of a spender.
+///
+/// # Parameters
+/// - `id` - The ID of the asset.
+/// - `spender` - The account that is allowed to spend the tokens.
+/// - `value` - The number of tokens to decrease the allowance by.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the operation fails.
+#[inline]
+pub fn decrease_allowance(id: AssetId, spender: AccountId, value: Balance) -> Result<()> {
+	build_dispatch(DECREASE_ALLOWANCE)
+		.input::<(AssetId, AccountId, Balance)>()
+		.output::<Result<()>, true>()
+		.handle_error_code::<StatusCode>()
+		.call(&(id, spender, value))
+}
+
+/// Creates `value` amount tokens and assigns them to `account`, increasing the total supply.
+///
+/// # Parameters
+/// - `id` - The ID of the asset.
+/// - `account` - The account to be credited with the created tokens.
+/// - `value` - The number of tokens to mint.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the operation fails.
+pub fn mint(id: AssetId, account: AccountId, value: Balance) -> Result<()> {
```

Comment:
this is missing #[inline] as the other ones also have.

Useful read on when to use inline: https://matklad.github.io/2021/07/09/inline-in-rust.html

Seems the optimized wasm doesn't have much difference, but the unoptimized wasm is larger without #[inline]

Reply:
Included in #190

Using it results into a contract of 10.6K, without 12.8K

The above results must not be compared with the 7.0K of the psp22 example in the ink! repo because it contains more functionality